base/commands/migration/migration_start.go

```diff
@@ -45,14 +53,34 @@ Selected data structures in the source cluster will be migrated to the target cl
 		}
 	}
 	ec.PrintlnUnnecessary("")
-	sts := NewStages(makeMigrationID(), ec.GetStringArg(argDMTConfig))
+	mID := MakeMigrationID()
+	sts := NewStartStages(ec.Logger(), mID, ec.GetStringArg(argDMTConfig))
 	sp := stage.NewFixedProvider(sts.Build(ctx, ec)...)
 	if _, err := stage.Execute(ctx, ec, any(nil), sp); err != nil {
 		return err
 	}
+	mStages, err := createMigrationStages(ctx, ec, ci, mID)
+	if err != nil {
+		return err
+	}
+	mp := stage.NewFixedProvider(mStages...)
+	_, err = stage.Execute(ctx, ec, any(nil), mp)
+	err2 := finalizeMigration(ctx, ec, ci, mID, ec.Props().GetString(flagOutputDir))
+	if err2 != nil {
+		return fmt.Errorf("finalizing migration: %w", err2)
+	}
```

Comment:
Should `finalizeMigration` be called even if `stage.Execute` panics?

Reply:
I think it should be called when there is an error executing `stage.Execute`. Member logs and migration report should be saved.

Reply:
Currently it won't run if `stage.Execute` panics, so I would defer `finalizeMigration` if it should be called in that case.